File: src/test/java/dev/retrotv/common/Log.java
package dev.retrotv.common;

import org.junit.jupiter.api.BeforeAll;

import java.util.logging.ConsoleHandler;
import java.util.logging.Logger;

public class Log {
    protected static final Logger log = Logger.getGlobal();

    @BeforeAll
    static void setLoggerFormatter() {
        log.setUseParentHandlers(false);
        LogFormatter formatter = new LogFormatter();
        ConsoleHandler handler = new ConsoleHandler();
        handler.setFormatter(formatter);
        log.addHandler(handler);
    }
}

File: src/test/kotlin/dev/retrotv/common/Log.kt
package dev.retrotv.common

import org.junit.jupiter.api.BeforeAll
import java.util.logging.ConsoleHandler
import java.util.logging.Logger

open class Log {
    companion object {

        @JvmStatic
        protected val log: Logger = Logger.getGlobal()

        @BeforeAll
        fun setLoggerFormatter() {
            log.setUseParentHandlers(false)
            val formatter = LogFormatter()
            val handler = ConsoleHandler()
            handler.setFormatter(formatter)
            log.addHandler(handler)
        }
    }
}
